pkg/apis/core/types_seed.go

```diff
@@ -108,7 +108,20 @@ type SeedBackup struct {
 type SeedDNS struct {
 	// IngressDomain is the domain of the Seed cluster pointing to the ingress controller endpoint. It will be used
 	// to construct ingress URLs for system applications running in Shoot clusters.
+	// If this is set spec.settings.nginxIngress can not be enabled and vise versa.
 	IngressDomain string
+	// NginxIngress controls settings for a gardener managed nginx-ingress controller
+	// +optional
+	NginxIngress *SeedSettingNginxIngress `json:"nginxIngress,omitempty" protobuf:"bytes,6,opt,name=nginxIngress"`
+}
+
+type SeedSettingNginxIngress struct {
+	// Enabled controls whether a gardener managed nginx-ingress is deployed. It is used to route external traffic to the Seed.
+	Enabled bool `json:"enabled" protobuf:"bytes,1,opt,name=enabled"`
```

Comment:
Do we need this field? If `SeedSettingNginxIngress` is not `nil`, then it's enabled, right?